files/report.rb

```diff
@@ -120,7 +120,7 @@ def metrics_to_hash(report)
       report_status["failed"] += 1
     end
     # fix for Puppet non-resource errors (i.e. failed catalog fetches before falling back to cache)
-    report_status["failed"] += report.logs.find_all {|l| l.source =~ /Puppet$/ && l.level.to_s == 'err' }.count
+    report_status["failed"] += report.logs.find_all {|l| l.level.to_s == 'err' && (l.source =~ /Puppet$/ || l.source == 'Facter') }.count
```

Comment:
I'd prefer to keep the diff minimal and not move things around:
```suggestion
    report_status["failed"] += report.logs.find_all {|l| (l.source =~ /Puppet$/ || l.source == 'Facter') && l.level.to_s == 'err' }.count
```

Reply:
No problem, changed.